scanamo/src/main/scala/org/scanamo/ops/ScanamoOpsA.scala

```diff
@@ -1,48 +1,80 @@
+    liftF[ScanamoOpsA, ScanamoResult[GetItemResult]](Get(req, identity))
+  def delete(
     req: ScanamoDeleteRequest
-  ): ScanamoOps[Either[ConditionalCheckFailedException, DeleteItemResult]] =
-    liftF[ScanamoOpsA, Either[ConditionalCheckFailedException, DeleteItemResult]](ConditionalDelete(req))
-  def scan(req: ScanamoScanRequest): ScanamoOps[ScanResult] = liftF[ScanamoOpsA, ScanResult](Scan(req))
-  def query(req: ScanamoQueryRequest): ScanamoOps[QueryResult] = liftF[ScanamoOpsA, QueryResult](Query(req))
+  ): ScanamoOps[ScanamoResult[DeleteItemResult]] =
+    liftF[ScanamoOpsA, ScanamoResult[DeleteItemResult]](Delete(req, identity))
+  def scan(req: ScanamoScanRequest): ScanamoOps[ScanResult] = liftF[ScanamoOpsA, ScanResult](Scan(req, identity))
+  def query(req: ScanamoQueryRequest): ScanamoOps[QueryResult] = liftF[ScanamoOpsA, QueryResult](Query(req, identity))
   def batchWrite(req: BatchWriteItemRequest): ScanamoOps[BatchWriteItemResult] =
-    liftF[ScanamoOpsA, BatchWriteItemResult](BatchWrite(req))
+    liftF[ScanamoOpsA, BatchWriteItemResult](BatchWrite(req, identity))
   def batchGet(req: BatchGetItemRequest): ScanamoOps[BatchGetItemResult] =
-    liftF[ScanamoOpsA, BatchGetItemResult](BatchGet(req))
-  def update(req: ScanamoUpdateRequest): ScanamoOps[UpdateItemResult] =
-    liftF[ScanamoOpsA, UpdateItemResult](Update(req))
-  def conditionalUpdate(
+    liftF[ScanamoOpsA, BatchGetItemResult](BatchGet(req, identity))
+  def update(
     req: ScanamoUpdateRequest
-  ): ScanamoOps[Either[ConditionalCheckFailedException, UpdateItemResult]] =
-    liftF[ScanamoOpsA, Either[ConditionalCheckFailedException, UpdateItemResult]](ConditionalUpdate(req))
+  ): ScanamoOps[ScanamoResult[UpdateItemResult]] =
+    liftF[ScanamoOpsA, ScanamoResult[UpdateItemResult]](Update(req, identity))
+//TODO: Does there need to be a fail lift?
+
+  implicit val functionScanamoOps = new Functor[ScanamoOpsA] {
+    override def map[A, B](fa: ScanamoOpsA[A])(f: A => B): ScanamoOpsA[B] = fa match {
+      case Put(req, res) => Put(req, res.andThen(f))
+      case Get(req, res) => Get(req, res.andThen(f))
+      case Delete(req, res) => Delete(req, res.andThen(f))
+      case Scan(req, res) => Scan(req, res.andThen(f))
+      case Query(req, res) => Query(req, res.andThen(f))
+      case BatchWrite(req, res) => BatchWrite(req, res.andThen(f))
+      case BatchGet(req, res) => BatchGet(req, res.andThen(f))
+      case Update(req, res) => Update(req, res.andThen(f))
+      case fail: Fail => fail
+    }
+  }
+
+
+  implicit val monadErrorScanamoOps = new MonadError[ScanamoOps, Throwable] {
+    def raiseError[A](e: Throwable): ScanamoOps[A] = liftF[ScanamoOpsA, A](Fail(e))
+
+    override def handleErrorWith[A](fa: ScanamoOps[A])(f: Throwable => ScanamoOps[A]): ScanamoOps[A] =
+      fa.resume match {
+        case Left(Fail(t)) => f(t)
+        case Left(_)  => f(new Throwable("Unexpected exception"))
```

Comment:
With resume, you either get another computation (`Left`), or you get the final result (`Right`). So here since we have a `Left(op)` where `op : ScanamoOpsA[ScanamoOps[A]]` and we need to return a `ScanamoOps[A]`, what you have to do is essentially put it back on the stack. You can do that with [`roll`](https://typelevel.org/cats/api/cats/free/Free$.html#roll[F[_],A](value:F[cats.free.Free[F,A]]):cats.free.Free[F,A]):
```scala
case Left(op) => Free.roll(op)
```